core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala

```diff
@@ -242,21 +260,28 @@ class BpmBillingProjectLifecycle(
           )
         case _ => ()
       }
-      _ <- billingRepository.getLandingZoneId(projectName).flatMap {
+      (jobControlId, eventType) <- billingRepository.getLandingZoneId(projectName).map {
         case Some(landingZoneId) =>
-          cleanupLandingZone(UUID.fromString(landingZoneId), projectName, ctx)
+          val result = cleanupLandingZone(UUID.fromString(landingZoneId), ctx)
+          (UUID.fromString(result.getJobReport.getId), AzureBillingProjectDelete)
         case None =>
-          logger.warn(s"Deleting BPM-backed billing project ${projectName}, but no associated landing zone to delete")
-          Future.successful()
-      }
-      _ <- billingRepository.getBillingProfileId(projectName).flatMap {
-        case Some(billingProfileId) =>
-          cleanupBillingProfile(UUID.fromString(billingProfileId), projectName, ctx)
-        case None =>
-          logger.warn(
-            s"Deleting BPM-backed billing project ${projectName}, but no associated billing profile record to delete"
-          )
-          Future.successful()
+          logger.warn(s"Deleting BPM-backed billing project $projectName, but no associated landing zone to delete")
+          (UUID.randomUUID(), OtherBpmBillingProjectDelete)
```

Comment:
Why return a random UUID here?

Reply:
The orchestrator needs a UUID so it can schedule the follow-up job.

If there wasn't a landing zone to delete, it still needs to schedule the job.

I think your point below or making it an `Option`, and then letting the orchestrator generate the UUID if it needs it is the way to go, though.